### ISSUE TITLE
932190 rule is causing FP

### ISSUE THREAD

**Issue by steamblade** [coreruleset/coreruleset#4213]
### Description

932190 rule is causing FP. The ? in this case was part of a valid query string (ts=23465), not a suspicious wildcard.

`curl -s "https://sandbox.coreruleset.org/test?url=https://test.abc.com/game?ts=23465" -H "x-format-output: txt-matched-rules" -H "x-crs-version: 4.8" -H "x-crs-paranoia-level: 3"`

To fix this, the regex need to be updated to `/(?:[?*]+[a-z/]+|[a-z/]+[?*]+(?![^=]*=))`, adding a negative lookahead to ensure it no longer matches when a ? or * is followed by an =—indicating a valid key=value pair.

### Confirmation

[ ] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by airween** [comment]
Hi @steamblade,

thanks for reporting this issue.

> The ? in this case was part of a valid query string (ts=23465), not a suspicious wildcard.

Yes, it's possible, especially if you use CRS on PL3.

Please take a look at the [documentation](https://github.com/coreruleset/coreruleset/blob/v4.16.0/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf#L1866):

_"..._
_# Please, keep in mind that this rule could lead to many false positives."_

Also please note that the problem here is not the `?`, but the wider expression - here is what I see in the log:
```
[data "Matched Data: /game? found within ARGS:url: https:/test.abc.com/game?st=23465"]
```
The point is that the report is about the string `/game?`, and not the `?` itself.

> To fix this, the regex need to be updated ..., adding a negative lookahead

Negative look-ahead (also the look-behind) is a PCRE/PCRE2 specific feature (of course there are many other regex engine supports it, like Python's, Java's...), and ModSecurity uses PCRE and PCRE2. But CRS is not only for ModSecurity, and it's not sure that those WAF's have a regex engine which supports negative look-ahead (and look-behind). Like re2, Hyperscan, etc.

So CRS definitely tries to avoid negative look-ahead regexes.

Summary:
* on PL3 these FP's are unpleasant but accepted (especially since we warn the user about this)
* CRS tries to avoid regexes with negative look-ahead/look-behind.

What you can do is:
* decrease the PL
  * maybe only to a specific URL
* create an exclusion against this FP

**Comment by 0xst3m** [comment]
thz for your feedback.
